castervoice/apps/ide/jetbrains.py

```diff
@@ -0,0 +1,142 @@
+        IDE.MOVE_LINE_UP: R(Key("as-up")) * Repeat(extra="n"),
+        IDE.MOVE_LINE_DOWN: R(Key("as-down")) * Repeat(extra="n"),
+        IDE.EXPAND_SELECTION: R(Key("c-w")) * Repeat(extra="n"),
+        IDE.AUTO_INDENT: R(Key("ca-i")),
+        "[<n>] before" : R(Key("c-left")) * Repeat(extra="n"),
+        "[<n>] after" : R(Key("c-right")) * Repeat(extra="n"),
+        "[<n>] befores": R(Key("cs-left")) * Repeat(extra="n"),
+        "[<n>] afters" : R(Key("cs-right")) * Repeat(extra="n"),
+        IDE.CLOSE_TAB_N_TIMES: R(Key("c-f4/%s" % DELAY)) * Repeat(extra="n"),
+        IDE.RUN_PROJECT: R(Key("cs-f10")),
+        "settings": R(Key("ca-s")),
+
+        # must be bound manually
+        IDE.DELETE_TO_LINE_START: R(Key("a-d,0")),
+        IDE.DELETE_TO_LINE_END: R(Key("a-d,$")),
+        # jet brains can only split horizontally or vertically
+        IDE.SPLIT_WINDOW_UP: R(Key("cs-s,h")),
+        IDE.SPLIT_WINDOW_DOWN: R(Key("cs-s,h")),
+        IDE.SPLIT_WINDOW_RIGHT: R(Key("cs-s,v")),
+        IDE.SPLIT_WINDOW_LEFT: R(Key("cs-s,v")),
+        IDE.SPLIT_MOVE_UP: R(Key("cs-s/%s,up"))* Repeat(extra="n"),
+        IDE.SPLIT_MOVE_DOWN: R(Key("cs-s,down"))* Repeat(extra="n"),
+        IDE.SPLIT_MOVE_RIGHT: R(Key("cs-s,right"))* Repeat(extra="n"),
+        IDE.SPLIT_MOVE_LEFT: R(Key("cs-s,left"))* Repeat(extra="n"),
+
+        # refactoring
+        IDE.REFACTOR: R(Key("cas-t")),
+        IDE.RENAME: R(Key("s-f6")),
+        IDE.INLINE: R(Key("ca-n")),
+        IDE.EXTRACT_METHOD: R(Key("ca-m")),
+        IDE.EXTRACT_VARIABLE: R(Key("ca-v")) * Repeat(extra="n"),
+        IDE.EXTRACT_FIELD: R(Key("ca-f")) * Repeat(extra="n"),
+        IDE.EXTRACT_CONSTANT: R(Key("ca-c")) * Repeat(extra="n"),
+        IDE.EXTRACT_PARAMETER: R(Key("ca-p")) * Repeat(extra="n"),
+
+        # window navigation
+        IDE.GO_TO_EDITOR: R(Key("escape")),
+        IDE.GO_TO_PROJECT_EXPLORER: R(Key("a-1")),
+        IDE.TOGGLE_TERMINAL: R(Key("a-f12")),
+    }
+    defaults = {"n": 1, "mim": ""}
+
+
+# ---------------------------------------------------------------------------
+
+context = AppContext(executable="idea", title="IntelliJ") \
+          | AppContext(executable="idea64", title="IntelliJ") \
+          | AppContext(executable="studio64") \
+          | AppContext(executable="pycharm")
+grammar = Grammar("IntelliJ + Android Studio + PyCharm", context=context)
+
+if settings.SETTINGS["apps"]["jetbrains"]:
+    if settings.SETTINGS["miscellaneous"]["rdp_mode"]:
+        control.nexus().merger.add_global_rule(JetbrainsRule())
+    else:
+        jet_brains_rule = JetbrainsRule(name="jet brains")
+        gfilter.run_on(jet_brains_rule)
+        grammar.add_rule(jet_brains_rule)
+
+        # You will need to change your terminal to bash under settings->tools
+        # C:\Program Files\Git\bin\bash.exe
+        git_rule = GitBashRule()
+        gfilter.run_on(git_rule)
+        grammar.add_rule(git_rule)
```

Comment:
@LexiconCode  thoughts on merging here? I think we can actually create a general git rule and merge it into most editors since most of them have built-in terminals.